charmcraft/deprecations.py

```diff
@@ -33,6 +33,8 @@
 # documentation)
 _DEPRECATION_MESSAGES = {
     "dn01": "Configuration keywords are now separated using dashes.",
+    "dn02": "A charmcraft.yaml configuration file is now required.",
+    "dn03": "Bases configuration is now required.",
```

Comment:
Meta question: is it really required if not erroring?

Also,

- do we want charmcraft init to add a charmcraft.yaml?
- do we want to specify what it is being set to as a default in the message or documentation entry?

Reply:
(1) yes, init should add a charmcraft.yaml, I can add that to the list
(2) it is in the documentation at the URL (https://discourse.charmhub.io/t/charmcraft-deprecation-notices/4652)

Reply:
Done (1) @ https://github.com/canonical/charmcraft/pull/415